cfgmgr/coppmgr.cpp

```diff
@@ -472,11 +548,58 @@ void CoppMgr::doCoppTrapTask(Consumer &consumer)
             /*Duplicate check*/
             if (conf_present &&
                 (trap_group == m_coppTrapConfMap[key].trap_group) &&
-                (trap_ids == m_coppTrapConfMap[key].trap_ids))
+                (trap_ids == m_coppTrapConfMap[key].trap_ids) &&
+                (is_always_enabled == m_coppTrapConfMap[key].is_always_enabled))
             {
                 it = consumer.m_toSync.erase(it);
                 continue;
             }
+
+            /* if always_enabled field has been changed */
+            if (conf_present &&
+                (trap_group == m_coppTrapConfMap[key].trap_group) &&
+                (trap_ids == m_coppTrapConfMap[key].trap_ids))
+            {
+                std::vector<FieldValueTuple> feature_fvs;
+                if (is_always_enabled == "true")
+                {
+                    /* if the value was changed from false to true,
+                    if the trap is not installed, install it.
+                    otherwise, do nothing. */
+
+                    m_coppAlwaysEnabledTraps.insert(key);
+                    if (m_coppTrapConfMap.find(key) == m_coppTrapConfMap.end())
+                    {
+                        addTrap(fvs, trap_ids, trap_group);
+                    }
+                }
+                else
+                {
+                    /* if the value was changed from true to false,
+                    check if there is a feature enabled.
+                    if no, remove the trap. is yes, do nothing. */
+                    m_coppAlwaysEnabledTraps.erase(key);
+
+                    auto cacheIt = m_featuresTableCache.find(key);
+                    if (cacheIt != m_featuresTableCache.end())
+                    {
+                        feature_fvs = m_featuresTableCache[key];
+                        for (auto i: feature_fvs)
+                        {
+                            if (fvField(i) == "state" && fvValue(i) == "enabled")
+                            {
+                                it = consumer.m_toSync.erase(it);
+                                continue;
```

Comment:
'continue' will be applicable for the immediately preceding for loop. So this logic will remove trap even if enabled. Can you double check?

Reply:
Done